### DISCUSSION TITLE
Access auth in plv8 function

### DISCUSSION THREAD

**Discussion by tbillington** [supabase/supabase#4555] [Questions]
In the docs there's a [guide to setting up and using plv8](https://supabase.com/docs/guides/database/extensions/plv8).

However, it doesn't mention how to access auth inside a plv8 function, and the methods for sql don't work (`auth.uid()` or `uid()`). I did see this answer for plpgsql but it doesn't seem to carry over to plv8 https://github.com/supabase/supabase/discussions/3554.

What are the options available? I can't use RLS because I need to gate the modification of column values and creation/deletion of them behind logic, hence the function.

**Comment by GaryAustin1** [comment]
Can you not execute sql from pvl8?  If so you might be able to do select * from auth.uid().
